src/tools/illink/src/linker/Linker/TypeReferenceEqualityComparer.cs

```diff
@@ -11,6 +11,9 @@ namespace Mono.Linker
     // Copied from https://github.com/jbevain/cecil/blob/master/Mono.Cecil/TypeReferenceComparer.cs
     internal sealed class TypeReferenceEqualityComparer : EqualityComparer<TypeReference>
     {
+        [Obsolete("Default will point to default object equality comparer. Use the constructor to create an instance with a resolver.")]
+        public static new object Default => throw new InvalidOperationException();
```

Comment:
Accidentally used `TypeReferenceEqualityComparer.Default`, which went to `ObjectEqualityComparer.Default` and required more debugging time than it should have. Adding this property to avoid doing that again.